malariagen_data/ag3.py

```diff
@@ -314,6 +320,482 @@ def _repr_html_(self):
+            If 'auto' let dask decide chunk size. If 'native' use native zarr
+            chunks. Also, can be a target size, e.g., '200 MiB'.
+
+        Returns
+        -------
+        d : dask.array.Array
+            An array of nucleotides giving the reference genome sequence for the
+            given contig.
+
+        """
+        region = self.resolve_region(region)
+
+        if region.contig in self.contigs:
+            d = super().genome_sequence(
+                region, inline_array=inline_array, chunks=chunks
+            )
+            return d
+
+        elif region.contig in ["2RL", "3RL"]:
+            genome = self.open_genome()
+            contigs = (
+                region.contig[0] + region.contig[1],
+                region.contig[0] + region.contig[1],
+            )
+
+            d = da.concatenate(
+                [
+                    da_from_zarr(
+                        genome[contig], inline_array=inline_array, chunks=chunks
+                    )
+                    for contig in contigs
+                ]
+            )
+
+        if region.start:
+            slice_start = region.start - 1
+        else:
+            slice_start = None
+        if region.end:
+            slice_stop = region.end
+        else:
+            slice_stop = None
+        loc_region = slice(slice_start, slice_stop)
+
+        return d[loc_region]
+
+    def genome_features(
+        self, region=None, attributes=("ID", "Parent", "Name", "description")
+    ):
+        """Access genome feature annotations.
+
+        Parameters
+        ----------
+        region: str or list of str or Region or list of Region
+            Contig name (e.g., "2L"), gene name (e.g., "AGAP007280"), genomic
+            region defined with coordinates (e.g., "2L:44989425-44998059") or a
+            named tuple with genomic location `Region(contig, start, end)`.
+            Multiple values can be provided as a list, in which case data will
+            be concatenated, e.g., ["3R", "3L"].
+        attributes : list of str, optional
+            Attribute keys to unpack into columns. Provide "*" to unpack all
+            attributes.
+
+        Returns
+        -------
+        df : pandas.DataFrame
+            A dataframe of genome annotations, one row per feature.
+
+        """
+        debug = self._log.debug
+
+        if region is None:
+            df = super().genome_features(region=region, attributes=attributes)
+            return df
+
+        elif region is not None:
+            region = self.resolve_region(region)
+            debug("normalise to list to simplify concatenation logic")
+            if isinstance(region, Region):
+                region = [region]
+
+            if all([r.contig in self.contigs for r in region]):
+                df = super().genome_features(region=region, attributes=attributes)
+                return df
+
+            elif all([r.contig in ["2RL", "3RL"] for r in region]):
```

Comment:
What if someone gives `region=["2RL", "X"]`?

A bit weird, but what if someone gives `region=["2RL", "3L"]`?

Reply:
Yeah the first case in particular, jesus. need to write tests. will try a slightly different approach